src/device-state.coffee

```diff
@@ -675,24 +676,28 @@ module.exports = class DeviceState extends EventEmitter
 		@applyUnblocker?()
 		return
 
-	triggerApplyTarget: ({ force = false, delay = 0, initial = false } = {}) =>
-		if @applyInProgress
-			if !@scheduledApply?
+	triggerApplyTarget: ({ force = false, delay = 0, initial = false, isFromApi = false } = {}) =>
+		if @applyInProgress?
+			if !@scheduledApply? || (isFromApi && @applyDelayInProgress)
 				@scheduledApply = { force, delay }
+				if isFromApi
+					# Cancel promise delay if call came from api to prevent waiting due to backoff
+					@applyInProgress.cancel()
```

Comment:
So this is a little of a weird one but I think you can avoid the need for cancellation by just doing something like:
```coffee
@applyInProgress = new Promise (resolve) =>
	setTimeout(resolve, delay)
	@cancelDelay = resolve
```
and then calling `@cancelDelay()` here

Reply:
Interesting approach, but I like it. It's more transparent than just cancelling the delayed promise. I'll implement that now.

Reply:
I changed @cancelDelay to reject since it would otherwise execute the .then chain, which executes the applyTargetState with the old force and initial parameters.
Please have a look.